File: src/tools/packet-sniffer/packet-sniffer.go
package main

import (
	"fmt"
	"log"
	"time"

	"github.com/google/gopacket"
	"github.com/google/gopacket/pcap"
)

func packetHandler(packet gopacket.Packet) {
	// Replace this with the packet processing logic you want
	fmt.Println(packet)
}

func getDefaultNetworkDevice() string {
	// Find all available network devices
	devices, err := pcap.FindAllDevs()
	if err != nil {
		log.Fatal(err)
	}

	// Return the name of the first network device (the default one)
	if len(devices) > 0 {
		return devices[0].Name
	}

	log.Fatal("No network devices found")
	return ""
}

func main() {
	// Get the default network device
	defaultDevice := getDefaultNetworkDevice()

	// Open the default network device for packet capture
	handle, err := pcap.OpenLive(defaultDevice, 1600, true, pcap.BlockForever)
	if err != nil {
		log.Fatal(err)
	}
	defer handle.Close()

	packetSource := gopacket.NewPacketSource(handle, handle.LinkType())

	fmt.Println("Starting packet sniffer on device:", defaultDevice)

	// Start processing packets
	for packet := range packetSource.Packets() {
		packetHandler(packet)
		
		time.Sleep(1 * time.Second)
	}
}

File: src/tools/packet-sniffer/packet-sniffer.py
import time
import logging

from scapy.all import sniff

def packet_handler(packet):
    # Replace this with the packet processing logic you want
    print(packet.summary())

def get_default_network_device():
    from scapy.arch import get_if_list

    # Find all available network devices
    devices = get_if_list()

    # Return the name of the first network device (the default one)
    if devices:
        return devices[0]

    raise ValueError("No network devices found")

def main():
    # Get the default network device
    default_device = get_default_network_device()

    print("Starting packet sniffer on device:", default_device)

    # Start sniffing packets on the default network device
    try:
        sniff(iface=default_device, prn=packet_handler)
    except KeyboardInterrupt:
        print("Packet sniffer stopped.")

if __name__ == "__main__":
    logging.getLogger("scapy.runtime").setLevel(logging.ERROR)
    main()
